### DISCUSSION TITLE
Grenade Launcher Improvements

### DISCUSSION THREAD

**Discussion by Mamuts123** [FakeFishGames/Barotrauma#12311] [Ideas and suggestions]
I think the grenade launcher is the most useless two-handed weapon in the game right now. The rifle performs it's job just as well, if not better, because it shoots grenades and rifle bullets while not having to take up your back slot. The only thing the launcher has going for it is a slightly faster fire rate. I think the simple improvement would be to make the grenade launcher able to hold multiple grenades and maybe even make it possible to store multiple types of grenades like the auto-shotgun can. If you were able to hold 4 or more stun grenades for incapacitating, and another 4 explosive grenades for killing, the weapon would be used way more and be way more effective. I also propose a hotkey which let's you switch between these 2 types quickly, maybe with a slight delay (the same would then be possible for the autoshotgun). 
To summarise :

-GL should hold more grenade capacity (at least 3, maybe 4 or 5)
-GL should have 2 slots for different grenades like the auto-shotgun
-add a hotkey for quickly switching betweem both ammo types

Thank you

**Comment by BiffW** [comment]
I would suggest its capacity be increased to 4×2，so that player could swap betwwen different grenades while not having to reload too often.

**Comment by BiffW** [comment]
> add a hotkey for quickly switching betweem both ammo types

This sounds cool. As a modder, I wish it supported multi-slot swapping if devs decide to implement it.
The definition should be like:
```
<ItemContainer capacity="3" AllowQuickSwap="true">
 <Containable items="grenadelikeitems" />
 <SubContainer AllowQuickSwap="False">
  <Containable items="otheritems" />
 </SubContainer>
</ItemContainer>
```
When player press the swap key, the 3 items should circle-swap their positions, same for container that can contain more items.
